code/controllers/subsystems/ticker.dm

```diff
@@ -0,0 +1,564 @@
+	'sound/music/halloween/halloween.ogg',\
+	'sound/music/halloween/ghosts.ogg'*/
+	'sound/music/space.ogg',\
+	'sound/music/traitor.ogg',\
+	'sound/music/title2.ogg',\
+	'sound/music/clouds.s3m',\
+	'sound/music/space_oddity.ogg') //Ground Control to Major Tom, this song is cool, what's going on?
+
+/datum/controller/subsystem/ticker/Initialize()
+	pregame_timeleft = config.pregame_time
+	send2mainirc("Server lobby is loaded and open at byond://[config.serverurl ? config.serverurl : (config.server ? config.server : "[world.address]:[world.port]")]")
+	return ..()
+
+/datum/controller/subsystem/ticker/fire(resumed = FALSE)
+	switch(current_state)
+		if(GAME_STATE_INIT)
+			pregame_welcome()
+			current_state = GAME_STATE_PREGAME
+		if(GAME_STATE_PREGAME)
+			pregame_tick()
+		if(GAME_STATE_SETTING_UP)
+			setup_tick()
+		if(GAME_STATE_PLAYING)
+			playing_tick()
+		if(GAME_STATE_FINISHED)
+			post_game_tick()
+
+/datum/controller/subsystem/ticker/proc/pregame_welcome()
+	to_chat(world, "<B><FONT color='blue'>Welcome to the pregame lobby!</FONT></B>")
+	to_chat(world, "Please set up your character and select ready. The round will start in [pregame_timeleft] seconds.")
+
+// Called during GAME_STATE_PREGAME (RUNLEVEL_LOBBY)
+/datum/controller/subsystem/ticker/proc/pregame_tick()
+	if(round_progressing && last_fire)
+		pregame_timeleft -= (world.time - last_fire) / (1 SECOND)
+
+	if(start_immediately)
+		pregame_timeleft = 0
+	else if(SSvote.time_remaining)
+		return // vote still going, wait for it.
+
+	// Time to start the game!
+	if(pregame_timeleft <= 0)
+		current_state = GAME_STATE_SETTING_UP
+		Master.SetRunLevel(RUNLEVEL_SETUP)
+		if(start_immediately)
+			fire() // Don't wait for next tick, do it now!
+		return
+
+	if(pregame_timeleft <= config.vote_autogamemode_timeleft && !SSvote.gamemode_vote_called)
+		SSvote.autogamemode() // Start the game mode vote (if we haven't had one already)
+
+// Called during GAME_STATE_SETTING_UP (RUNLEVEL_SETUP)
+/datum/controller/subsystem/ticker/proc/setup_tick(resumed = FALSE)
+	if(!setup_choose_gamemode())
+		// It failed, go back to lobby state and re-send the welcome message
+		pregame_timeleft = config.pregame_time
+		SSvote.gamemode_vote_called = FALSE // Allow another autogamemode vote
+		current_state = GAME_STATE_PREGAME
+		Master.SetRunLevel(RUNLEVEL_LOBBY)
+		pregame_welcome()
+		return
+	// If we got this far we succeeded in picking a game mode.  Punch it!
+	setup_startgame()
+	return
+
+// Formerly the first half of setup() - The part that chooses the game mode.
+// Returns 0 if failed to pick a mode, otherwise 1
+/datum/controller/subsystem/ticker/proc/setup_choose_gamemode()
+	//Create and announce mode
+	if(master_mode == "secret")
+		src.hide_mode = TRUE
+
+	var/list/runnable_modes = config.get_runnable_modes()
+	if((master_mode == "random") || (master_mode == "secret"))
+		if(!runnable_modes.len)
+			to_chat(world, "<B>Unable to choose playable game mode.</B> Reverting to pregame lobby.")

```

Comment:
`to_world()`? Filtering tags? Leaving this as the last comment for this because there's several other to_chat()s below here and leaving a review with 20 comments complaining about the same thing in the same file feels tedious and a little rude